File: campus-navigate-app/flutter_app/lib/domain/pdr/algorithms/mock_motion_batch.dart
import 'package:indoor_navigation/domain/pdr/pdr_models.dart';

List<MotionInputSample> createMockMotionBatch({
  required double desiredHeadingDegrees,
  required int nowMs,
}) {
  final headings = <double>[
    desiredHeadingDegrees - 6,
    desiredHeadingDegrees + 4,
    desiredHeadingDegrees + 9,
    desiredHeadingDegrees - 3,
  ];

  return List.unmodifiable(
    List.generate(headings.length, (index) {
      return MotionInputSample(
        acceleration: MotionVector(x: index == 1 ? 0.4 : 1.45, y: 0, z: 0),
        headingDegrees: headings[index],
        timestampMs: nowMs - (headings.length - index) * 18,
      );
    }),
  );
}

File: tarumt-nav-app/expo-app/src/pdr/model/mockMotionBatch.ts
import type { MotionInputSample } from '../type';

export function createMockMotionBatch(input: {
  desiredHeadingDegrees: number;
  nowMs: number;
}) {
  const headings = [
    input.desiredHeadingDegrees - 6,
    input.desiredHeadingDegrees + 4,
    input.desiredHeadingDegrees + 9,
    input.desiredHeadingDegrees - 3,
  ];

  return headings.map(
    (headingDegrees, index): MotionInputSample => ({
      acceleration: {
        x: index === 1 ? 0.4 : 1.45,
        y: 0,
        z: 0,
      },
      headingDegrees,
      timestampMs: input.nowMs - (headings.length - index) * 18,
    }),
  );
}
